### PR TITLE
Prepare release 2.9.0.0

### PR DESCRIPTION
**Pull request by fendor**
# Releasing

## Release checklist

- [x] check ghcup supports new GHC releases if any
- [x] set the supported GHCs in workflow file `.github/workflows/release.yaml`
  - There is currently a list of GHC versions for each major platform. Search for `ghc: [` to find all lists.
  - Look for `TODO:` to find locations that require extra care for GHC versions.
- [x] check all plugins still work if release includes code changes
- [x] bump package versions in all `*.cabal` files (same version as hls)
  - HLS uses lockstep versioning. The core packages and all plugins use the same version number, and only support exactly this version.
    - Exceptions:
      - `hie-compat` requires no automatic version bump.
      - `shake-bench` is an internal testing tool, not exposed to the outside world. Thus, no version bump required for releases.
  - For updating cabal files, the following script can be used:
    - ```sh
      ./release/update_versions.sh <OLD_VERSION> <NEW_VERSION>
      ```
    - It still requires manual verification and review
- [x] generate and update changelog
  - Generate a ChangeLog via `./GenChangelogs.hs <api-key> <tag>`
    - `<tag>` is the git tag you want to generate the ChangeLog from.
    - `<api-key>` is a github access key: https://github.com/settings/tokens
- [x] create release branch as `wip/<version>`
  - `git switch -c wip/<version>`
- [x] create release tag as `<version>`
  - `git tag <version>`
- [x] trigger release pipeline by pushing the tag
  - this creates a draft release
  - `git push <remote> <version>`
- [x] run `sh scripts/release/download-gh-artifacts.sh <version> <your-gpg-email>`
  - downloads artifacts to `gh-release-artifacts/haskell-language-server-<version>/`
  - also downloads FreeBSD bindist from circle CI
  - adds signatures
- [x] upload artifacts to downloads.haskell.org from `gh-release-artifacts/haskell-language-server-<version>/`
  - You require sftp access, contact wz1000, bgamari or chreekat
  - `cd gh-release-artifacts/haskell-language-server-<version>`
  - `SIGNING_KEY=... ../../release/upload.sh upload`
    - Your SIGNING_KEY can be obtained with `gpg --list-secret-keys --keyid-format=long`
  - Afterwards, the artifacts are available at: `https://downloads.haskell.org/~hls/haskell-language-server-<version>/`
  - Run `SIGNING_KEY=... ../../release/upload.sh purge_all` to remove CDN caches
- [x] create PR to [ghcup-metadata](https://github.com/haskell/ghcup-metadata)
  - [x] update `ghcup-vanilla-0.0.8.yaml` and `ghcup-vanilla-0.0.7.yaml`
    - can use `sh scripts/release/create-yaml-snippet.sh <version>` to generate a snippet that can be manually inserted into the yaml files
  - ~~update `hls-metadata-0.0.1.json`~~ Currently unnecessary, GHCup builds its own HLS binaries and updates that file.
    - utilize `cabal run ghcup-gen -- generate-hls-ghcs -f ghcup-0.0.7.yaml --format json --stdout` in the root of ghcup-metadata repository
  - Be sure to mark the correct latest version and add the 'recommended' tag to the latest release.
- [ ] get sign-off on release
  - from wz1000, michealpj, maerwald and fendor
- [ ] publish release on github
- [ ] upload hackage packages
  - requires credentials
- [x] update https://haskell-language-server.readthedocs.io/en/latest/support/ghc-version-support.html#current-ghc-version-support-status
- [x] Supported tools table needs to be updated:
  - https://www.haskell.org/ghcup/install/#supported-platforms
  - https://github.com/haskell/ghcup-hs/blob/master/docs/install.md#supported-platforms
  - https://github.com/haskell/ghcup-metadata/blob/44c6e2b5d0fcae15abeffff03e87544edf76dd7a/ghcup-gen/Main.hs#L67
- [ ] post release on discourse and reddit
- [ ] merge release PR to master or forward port relevant changes

### CODE REVIEW

**Comment by michaelpj** [comment]
I think we should at least wait for https://github.com/haskell/haskell-language-server/pull/4314 and re-enable the fourmolu plugin.

**Comment by fendor** [comment]
@michaelpj Ok, that's fine by me. If we wait, we can also include #4318. I am just making sure, we are not further bogged down by CI issues.

**Comment by michaelpj** [comment]
Remember to update the support tables in the docs (that should be in the checklist!)

**Comment by fendor** [comment]
It is in the checklist, just the fourth to last item, which is clearly too late :) Moved it up in the checklist.

**Comment by fendor** [comment]
Done, updated the supported GHC version table

**Comment by michaelpj** [comment]
Uh... I should have actually read the checklist before commenting :sweat_smile:

File: docs/support/plugin-support.md [hunk 1]
```diff
@@ -64,6 +64,6 @@ For example, a plugin to provide a formatter which has itself been abandoned has
 | `hls-overloaded-record-dot-plugin`  | 2    |                          |
 | `hls-semantic-tokens-plugin`        | 2    |                          |
 | `hls-floskell-plugin`               | 3    |                          |
-| `hls-stan-plugin`                   | 3    | 9.2.(4-8)                |
+| `hls-stan-plugin`                   | 3    | 9.2.(4-8), 9.10.1        |
 | `hls-retrie-plugin`                 | 3    |                          |
```
Comment (michaelpj) [inline docs/support/plugin-support.md:68]: There are more plugins not supported on 9.10!

Comment (fendor) [reply docs/support/plugin-support.md:68]: Which ones? In `haskell-language-server.cabal`, I could only find stan having:

```
common stan
  if flag(stan) && (impl(ghc > 8.8.1) && impl(ghc <= 9.2.3) || impl(ghc >= 9.4.0) && impl(ghc < 9.10.0))
    build-depends: haskell-language-server:hls-stan-plugin
    cpp-options: -Dhls_stan

...

  if flag(stan) && (impl(ghc > 8.8.1) && impl(ghc <= 9.2.3) || impl(ghc >= 9.4.0) && impl(ghc < 9.10.0))
    buildable: True
  else
    buildable: False
```

Comment (fendor) [reply docs/support/plugin-support.md:68]: Ah, I found more in `.github/workflow/test.yml`

Comment (michaelpj) [reply docs/support/plugin-support.md:68]: Hmm, I'm surprised nothing else has guards in the cabal file? I would expect us to still try and build them if not?

Comment (fendor) [reply docs/support/plugin-support.md:68]: Does that work? Like will this not affect `cabal install haskell-language-server -w 9.10.1`?

Comment (michaelpj) [reply docs/support/plugin-support.md:68]: I'm unsure what our policy should be here. We should probably either be consistently setting things in `cabal.project` or in the cabal file...

**Comment by michaelpj** [comment]
I removed the remaining unnecesssary cabal.project bits

**Comment by michaelpj** [comment]
Reverted sorry, I was doing it wrong

**Review by michaelpj — APPROVED** [review]

**Review by jhrcek — APPROVED** [review]
LGTM

**Review by soulomoon — APPROVED** [review]
LGTM！

**Review by wz1000 — APPROVED** [review]

**Comment by benz0li** [comment]
ℹ️ One has to patch https://github.com/haskell/haskell-language-server/blob/287ee42ab566e5c088d80ec859d991f5be04f66e/GNUmakefile#L141 when using cabal-install versions ≥ 3.11 (i.e. [cabal-install-v3.12.0.0-prerelease](https://github.com/haskell/cabal/releases/tag/cabal-install-v3.12.0.0-prerelease)).  
👉 Patch: Modify `"$(STORE_DIR)/ghc-$(GHC_VERSION)"` to `"$(STORE_DIR)/ghc-$(GHC_VERSION)-inplace"`.

Cross reference:

* https://github.com/benz0li/ghc-musl/commit/77ae6ed566a2c41d95827928567a5a866016be30

**Pull request merged**